### PR TITLE
f08: add elemental to eq/neq operator functions

### LINKED ISSUE #5875: fortran: missing ELEMENTAL in eq/neq operator functions
Please see discussion at 
https://github.com/open-mpi/ompi/issues/10057
and 
https://lists.mpi-forum.org/pipermail/mpiwg-fortran/2022-March/003613.html

the same issue applies to MPICH (at least the versions I tried 3.4.1  from Fedora packaging, 3.4.3 from source, have not yet tried the latest)

Cheers
Salvatore

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
Adding elemental allows user to apply the operator to scalar/array
comparisons.

Fixes #5875 
[skip warnings]

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch3/tcp
test:mpich/ch4/ofi

**Comment by hzhou** [comment]
NOTE: wait for feedback from MPI forum before merging. Ref: https://lists.mpi-forum.org/pipermail/mpiwg-fortran/2022-March/003613.html

**Comment by hzhou** [comment]
Removing the "do-no-merge" label as the discussion in https://github.com/open-mpi/ompi/issues/10057# has resolved.

**Review by raffenet — APPROVED** [review]

**Pull request merged**
